### DISCUSSION TITLE
Direct printing without visual presentation

### DISCUSSION THREAD

**Discussion by b-mi** [MicrosoftEdge/WebView2Feedback#4363] [Q&A] [answered]
How to print a pdf file to a printer in WebView2 without a visual display of the pdf in WebView2?

**Comment by victorhuangwq** [comment]
@monica-ch I don't think this would be currently possible? If I am not wrong, there is no "silent" printing right?

**Reply by monica-ch** [reply, marked as answer]
@b-mi  This can be done using WebView2 Print API that prints a pdf file without showing it to the end user

https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/print?tabs=dotnetcsharp

**Comment by b-mi** [comment]
There is support for silent printing of the current web page. So there is support for silent printing of DISPLAYED content. But the question asks that I have a pdf file, and I need to print it silently via webview2. WITHOUT VIEWING that pdf. Actually, it is about printing via webview2 without displaying webview2.

**Comment by monica-ch** [comment]
Ah, this is not currently supported in WebView2. Please open a feature request [here]( https://github.com/MicrosoftEdge/WebView2Feedback/issues/new/choose)
